Docs/INISystem.md

```diff
@@ -1,230 +1,403 @@
 
 ### User Defined Constants
-
-```
+```ini
 MY_EXAMPLE_CONSTANT=15
 ```
 The above user-defined or system constants can be used elsewhere as:
-```
+```ini
 [MyExampleControl]
 $X=MY_EXAMPLE_CONSTANT
 ```
 _NOTE: Constants can only be used in [dynamic control properties](#dynamic-control-properties)_
 
+### Data Types
+- The `text` use `@` as a line break. To write the real `@` character, use `\@`. Also as INI syntax uses `;` to denote comments, use `\semicolon` to write the real `;` character.
+- The `color string` use form `R,G,B` or `R,G,B,A`. All values must be between `0` and `255`.
+- The `boolean` string value parses as `true` if it contains one of these symbol as first character: `t`, `y`, `1`, `a`, `e`; and if first symbol is `n`, `f`, `0`, then it parses as `false`. 
+- The `integer` type is actually `System.Int32`.
+- The `float` type is actually `System.Single`.
+- The `N integers` or `N floats` is a `integer` or `float` type values repeated `N` times, but separated with `,` character without spaces e.g., `0,0` or `0.0,0.0` respectively.
+- The `comma-separated strings` is a string, but separated with `,` character without spaces e.g., `one,two,three`.
+<!-- - The `comma separated integers` or `comma separated floats` is a `integer` or `float` type, but separated with `,` character without spaces e.g., `0,0` or `0.0,0.0` respectively. -->
+
 ## Control properties
 Below lists basic and dynamic control properties. Ordering of properties is important. If there is a property that relies on the size of a control, the properties must set the size of that control first.
 
-> **Note**
-> Multi-line strings use `@` as a line break. To write the real `@` character, use `\@`. Also as INI syntax uses `;` to denote comments, use `\semicolon` to write the real `;` character.
-
 ### Basic Control Properties
 Basic control properties cannot use constants.
+> [!WARNING]
+> Do not copy-paste ini-code below without edits because it won't work! It shows only how to work with properties.
+> 
+> For example,
+> - `X` and `Y` are conflict with `Location`,
+> - `BackgroundTexture` and `SolidColorBackgroundTexture` conflicts,
+> - and many others.
+
+#### [XNAControl](https://github.com/Rampastring/Rampastring.XNAUI/blob/master/XNAControls/XNAControl.cs)
+- Basic class inherited by any other control element.
+```ini
+[SOMECONTROL]              ; XNAControl
+X=0                        ; integer,      the X location of the control.
+Y=0                        ; integer,      the Y location of the control.
+Location=50,100            ; 2 integers,   the X and Y location of the control.
+Width=25                   ; integer,      the Width of the control.
+Height=10                  ; integer,      the Height of the control.
+Size=60,30                 ; 2 integers,   the Width and Height of the control.
+Text=Some Text@Example     ; text,         the text to display for the control (ex: buttons, labels, etc...).
```

Comment:
The values should contain default values or no values, if there is no default value. This will ensure that a config maker could copy-paste the reference INI code and have nothing changed in the behavior